options.go

```diff
@@ -369,12 +376,23 @@ func HTTPServer(server *http.Server) RouterOption {
 // DocsHandler sets the documentation rendering handler function. You can
 // use `huma.RapiDocHandler`, `huma.ReDocHandler`, `huma.SwaggerUIHandler`, or
 // provide your own (e.g. with custom auth or branding).
+//
+// DEPRECATED! Use `DocsDomType` instead!
 func DocsHandler(f Handler) RouterOption {
+	fmt.Println("This option is deprecated, use `DocsDomType` instead")
```

Comment:
Can this be supported without deprecating it?

Reply:
Strictly speaking, yes.  Where it gets weird is if you use the `DocsRoutePrefix` RouterOption to change the prefix of the where the docs live.

E.g.
```
NewServer("api", "version", DocsRoutePrefix("/prefix"), DocsHandler(ReDocHandler(title))))
```

Since `DocsHandler` is just setting a function to call when `/docs` is hit, and `ReDocHandler` isn't aware of the new prefix, it felt weird to have both `DocsHandler`s and `DocsRoutePrefix`es.

A way to get around that would be to call:
```
NewServer("api", "version", DocsRoutePrefix("/prefix"), DocsDomType("redoc")))
```

which would give the user the ability to use ReDoc, while also ensuring that the docs are reachable at `/prefix/docs`.

Which is why I went for the deprecation route.